### PR TITLE
Add eligibility checker on pages

### PR DESCRIPTION
**Pull request by loevgaard**
Related #17 

image

### CODE REVIEW

File: src/Checker/Eligibility/Page/EnabledDateEligibilityChecker.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace Setono\SyliusCMSPlugin\Checker\Eligibility\Page;
+
+use Setono\SyliusCMSPlugin\Model\PageInterface;
+
+final class EnabledDateEligibilityChecker implements EligibilityCheckerInterface
+{
+    public function isEligible(PageInterface $page): bool
+    {
+        return $page->isEnabledOn(new \DateTimeImmutable());
```
Comment (Roshyo) [inline src/Checker/Eligibility/Page/EnabledDateEligibilityChecker.php]: We should import this `DateTimeImmutable`

File: src/Model/EnabledDateIntervalAwareTrait.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        if (null === $this->enabledUntil && $dateTime >= $this->enabledFrom) {
+            return true;
+        }
+
+        if ($dateTime >= $this->enabledFrom && $dateTime <= $this->enabledUntil) {
+            return true;
+        }
+
+        return false;
```
Comment (Roshyo) [inline src/Model/EnabledDateIntervalAwareTrait.php]:
```suggestion
        if (null !== $this->enabledUntil && $this->enabledUntil < $dateTime) {
            return false;
        }
        if (null !== $this->enabledFrom && $this->enabledFrom > $dateTime) {
            return false;
        }

        return true;
```

File: src/Model/Page.php [hunk 3]
```diff
@@ -25,8 +30,16 @@ class Page implements PageInterface, TranslatableInterface
 
     protected ?ViewInterface $view = null;
 
+    /**
+     * @psalm-var Collection<array-key, BaseChannelInterface>
+     *
+     * @var Collection|BaseChannelInterface
```
Comment (Roshyo) [inline src/Model/Page.php]: Shouldn't it be `Collection<array-key, BaseChannelInterface>` instead ?

File: src/DependencyInjection/Compiler/RegisterPageEligibilityCheckersPass.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    public function process(ContainerBuilder $container): void
+    {
+        if (!$container->has('setono_sylius_cms.checker.eligibility.page.composite')) {
+            return;
+        }
+
+        $checker = $container->getDefinition('setono_sylius_cms.checker.eligibility.page.composite');
+
+        /** @var string $id */
+        foreach (array_keys($container->findTaggedServiceIds('setono_sylius_cms.page_eligibility_checker')) as $id) {
```
Comment (Roshyo) [inline src/DependencyInjection/Compiler/RegisterPageEligibilityCheckersPass.php:23]: import `array_keys`

**Review by Roshyo — CHANGES_REQUESTED** [review]
:+1: Good job

File: src/Model/Page.php (hunk 3 above)
Comment (loevgaard) [reply src/Model/Page.php]: There was a time auto completion in PhpStorm didn't work with that, but I will try

Comment (Roshyo) [reply src/Model/Page.php]: At worst, `var Collection|BaseChannelInterface[]` 

But from what I've seen, PhpStorm autocompletion now works with the first syntax

Comment (loevgaard) [reply src/Model/Page.php]: Still doesn't work, but I still fixed an auto completion bug

**Review by Roshyo — APPROVED** [review]

**Pull request merged**